### PR TITLE
Override DeleteNamespaceNamespaceDeleteDelay config in DeleteNamespace API

### PR DESCRIPTION
**Pull request by hehaifengcn**
## What changed?

Provides an option to control when namespace metadata will be deleted.

## Why?

Currently `DeleteNamespace` workflow first deletes workflow history then use `frontend.deleteNamespaceNamespaceDeleteDelay` dynamic config to delay the deletion of namespace metadata. This change provides an option to config `NamespaceDeleteDelay` (so namespace metadata can be deleted immediately if needed). 

## How did you test it?

Add functional test. 

## Potential risks

No risks.

## Is hotfix candidate?

No.

### CODE REVIEW

File: tests/namespace_delete.go [hunk 1]
```diff
@@ -160,6 +160,64 @@ func (s *namespaceTestSuite) Test_NamespaceDelete_Empty() {
...
+
+	namespaceExistsOp := func() error {
+		_, err := s.frontendClient.DescribeNamespace(ctx, &workflowservice.DescribeNamespaceRequest{
+			Id: nsID,
+		})
+		var notFound *serviceerror.NamespaceNotFound
+		if errors.As(err, &notFound) {
+			return nil
+		}
+		return errors.New("namespace still exists")
```
Comment (hehaifengcn) [inline tests/namespace_delete.go:210]: Wonder why it triggered lint error as same code was used in other places in this file.

**Comment by tdeebswihart** [comment]
Would you mind filling out the PR body with _why_ you're making this change? I've no context as for why you're doing it or if there are any risks, etc.

**Comment by hehaifengcn** [comment]
Description added

File: tests/namespace_delete.go [hunk 2]
```diff
@@ -160,6 +160,64 @@ func (s *namespaceTestSuite) Test_NamespaceDelete_Empty() {
...
+			return nil
+		}
+		return errors.New("namespace still exists") // nolint:goerr113
+	}
+
+	namespaceExistsPolicy := backoff.NewExponentialRetryPolicy(time.Second).
+		WithBackoffCoefficient(1).
+		WithExpirationInterval(30 * time.Second)
+
+	err = backoff.ThrottleRetry(namespaceExistsOp, namespaceExistsPolicy, func(_ error) bool { return true })
```
Comment (alexshtin) [inline tests/namespace_delete.go:217]: Please replace it with `s.Eventually` (there are examples in this file). Then you won't need dynamic error, and can just return `true` or `false`.

File: tests/namespace_delete.go [hunk 3]
```diff
@@ -191,24 +234,17 @@ func (s *namespaceTestSuite) Test_NamespaceDelete_Empty_WithID() {
 	})
 	s.NoError(err)
 	s.Equal(enumspb.NAMESPACE_STATE_DELETED, descResp2.GetNamespaceInfo().GetState())
-
-	namespaceExistsOp := func() error {
+	s.Eventually(func() bool {
```
Comment (alexshtin) [inline tests/namespace_delete.go:237]: :+1:

**Review by alexshtin — APPROVED** [review]

**Pull request merged**